### DISCUSSION TITLE
Resolving External Screen Overlapping Issue

### DISCUSSION THREAD

**Discussion by emilehiot** [Jean-Tinland/simple-bar#382] [Q&A]
### Introduction

Many users have encountered an overlapping issue when using an external screen with the simple-bar. This tutorial aims to guide you through the steps to resolve this issue by modifying the `.yabairc`  configuration file.

### Step-by-Step Guide

1. **Open the `.yabairc` Configuration File**

   Navigate to your `.yabairc` file, which is usually located in your home directory.

2. **Add the Configuration Line**

   Add the following line to your `.yabairc` file to create a void space at the top of all spaces, which can accommodate a bar (or other elements, but in this case, it's for the bar):

```sh 
yabai -m config external_bar all:28:0
```

Note: The value `28` in `28:0` represents the number of pixels dedicated to the space. You can modify this value according to your preference.

3. **Save Your File**

After adding the line, save your `.yabairc` file to apply the changes.

4. **Restart the Configuration**

Restart the configuration to apply the new settings and check if the issue has been resolved. You can do this by using the following command in your terminal:

``` sh 
yabai --restart-service
```

### Yabai Documentation Excerpt

Here is an excerpt from the yabai documentation that further explains the `external_bar` configuration:

```
external_bar [<main|all|off>:<top_padding>:<bottom_padding>]

Specify top and bottom padding for a potential custom bar that you may be running.
- main: Apply the given padding only to spaces located on the main display.
- all: Apply the given padding to all spaces regardless of their display.
- off: Do not apply any special padding.
```

**Comment by Jean-Tinland** [comment]
I'd like to move this piece of information in the documentation (where I moved everything else). Is that okay for you?

Thank you by the way!

**Comment by Jean-Tinland** [comment]
I moved this here: https://www.jeantinland.com/en/toolbox/simple-bar/documentation/resolving-external-screen-overlapping-issue/.

I'm closing this, feel free to update this guide and notify me if necessary, thank you again :)

**Comment by liemle3893** [comment]
Not sure right place to ask. But does this requires SIP to be disabled? I have follow the docs but the app on External Screen got Overlapped.
